### DISCUSSION TITLE
How have you configured indentation for TypeScript/JavaScript files?

### DISCUSSION THREAD

**Discussion by cherrot** [ayamir/nvimdots#999] [Q&A] [answered]
There are 2 ways to configure indentation when editing files: through `set autoindent` or setting `indent = {enable = true}` in `nvim-treesitter.configs`. 

Neither of them could generate the same indentation which `prettier` does :(

So I'm curious how have you guys configured your indentation rules?  Is there a good reason that we didn't enable treesitter's indent by default?

Thanks in advance!

**Comment by charliie-dev** [comment, marked as answer]
dont we already have prettier formatter installed? why dont just use prettier?

we let formatter to deal with all that kind of things to keep related things in one place.

**Comment by cherrot** [comment]
In my experience `prettier` only works on saving a file (Please correct me if I'm wrong). I want auto-indent my code when I'm typing, otherwise I have to input lots of tabs/spaces :smile:

**Reply by charliie-dev** [reply]
ys, that's intended for this repo. auto-correction during editing is actually quite annoy and add lots of system overhead, causing u to feel leggy while editting.

If u want to change that, which is fairly simple, we had define some custom events in [this file](https://github.com/ayamir/nvimdots/blob/main/lua/modules/configs/completion/formatting.lua). just check what event u need and add new event trigger to it.

if u have no idea what i'm talking about, here's some hints:
1. search for `nvim custom autocmd`
2. search for `nvim event trigger`

**Comment by cherrot** [comment]
I agree with that. So far my solution is adding `shiftwidth=2` for js/jsx/ts/tsx files with `autoindent` being on (which is the default).

e.g. 

```lua
-- /lua/user/event.lua
local definitions = {
        ft = {
               { "FileType", "javascript", "set shiftwidth=2" },
               { "FileType", "javascriptreact", "set shiftwidth=2" },
               { "FileType", "typescript", "set shiftwidth=2" },
               { "FileType", "typescriptreact", "set shiftwidth=2" },
        }
```

**Reply by charliie-dev** [reply]
gd, but if u want to use 	`prettier` to auto-format your js/ts files, try the method i mentioned above.
